src/app/auth/session-selection/session-selection.component.ts

```diff
@@ -56,9 +56,15 @@ export class SessionSelectionComponent implements OnInit {
     /**
      * Persist repo information in local browser storage
      * To retrieve after authentication redirects back to WATcher
+     *
+     * Since localStorage::setItem with an undefined value can result in
+     * the subsequent value being stored as a string being 'undefined', check
+     * if undefined before storing it.
      */
-    window.localStorage.setItem('org', repoOrg);
-    window.localStorage.setItem('dataRepo', repoName);
+    if (repoOrg && repoName) {
+      window.localStorage.setItem('org', repoOrg);
+      window.localStorage.setItem('dataRepo', repoName);
+    }
```

Comment:
For the `else` case, we should clear the `localStorage` for `'org'` and `'dataRepo'` so that the values from the old sessions will not be used.